src/client/hud.cpp

```diff
@@ -862,6 +862,58 @@ void Hud::drawSelectionMesh()
 	}
 }
 
+void Hud::toggleBlockBounds()
+{
+	m_block_bounds_mode++;
+
+	if (m_block_bounds_mode > 2) {
+		m_block_bounds_mode = 0;
+	}
+}
+
+void Hud::drawBlockBounds()
+{
+	if (!m_block_bounds_mode) {
+		return;	
+	}
+
+	video::SMaterial old_material = driver->getMaterial2D();
+	driver->setMaterial(m_selection_material);
+
+	auto pos = player->getStandingNodePos();
+
+	auto blockPos = v3s16(
+		floorf((float) pos.X / 16.0f),
+		floorf((float) pos.Y / 16.0f),
+		floorf((float) pos.Z / 16.0f)
+	);
+
+	auto offset = intToFloat(client->getCamera()->getOffset(), BS);
+
+	s8 radius = 2;
+
+	auto halfNode = v3f(BS, BS, BS) / 2.0f;
+
+	for (s8 x = -radius; x <= radius; x++) {
+		for (s8 y = -radius; y <= radius; y++) {
+			for (s8 z = -radius; z <= radius; z++) {
+				auto blockOffset = v3s16(x, y, z);
+
+				if (blockOffset.getLength() == 0 || m_block_bounds_mode > 1) {
+					auto box = aabb3f(
+						intToFloat((blockPos + blockOffset) * 16, BS) - offset - halfNode,
+						intToFloat(((blockPos + blockOffset) * 16) + 15, BS) - offset + halfNode
+					);
```

Comment:
Please make use of `MAP_BLOCKSIZE` or `MAP_BLOCKSIZE-1` respectively.
You might also be interested in the function `getNodeBox()` (numeric.h)  which does a quite similar thing.